catalog/spatial/wfs/spatial-wfs-metacardtype-registry/src/main/java/org/codice/ddf/spatial/ogc/wfs/catalog/metacardtype/registry/impl/WfsMetacardTypeRegistryImpl.java

```diff
@@ -81,7 +82,24 @@ public void registerMetacardType(
   }
 
   /** {@inheritDoc} */
-  public void clear() {
+  @Override
+  public void clear(String sourceId) {
+    Verify.verifyNotNull(sourceId, "argument 'sourceId' may not be null.");
+    for (Iterator<ServiceRegistration<MetacardType>> iter = serviceRegistrations.iterator();
+        iter.hasNext(); ) {
+      ServiceRegistration registration = iter.next();
+      if (registration.getReference() != null
+          && registration.getReference().getProperty((SOURCE_ID)) != null
+          && registration.getReference().getProperty(SOURCE_ID).equals(sourceId)) {
```

Comment:
✏️ You already verify that `sourceId` is not null.  You could remove the previous line if you do `sourceId.equals(registration.getReference().getProperty(SOURCE_ID))`